java/src/main/kotlin/io/spine/validation/java/expression/EmptyFieldSpecification.kt

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2025, TeamDev. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.validation.java.expression
+
+import io.spine.protodata.ast.Field
+import io.spine.protodata.java.Expression
+import io.spine.protodata.java.JavaValueConverter
+import io.spine.protodata.java.field
+import io.spine.validation.UnsetValue
+import io.spine.validation.java.ValidationCodeInjector.MessageScope.message
+
+/**
+ * Provides an expression to check if a given [Field] holds
+ * the default (unset) value.
+ *
+ * It takes a Protobuf instance of a default value for the field type
+ * with [UnsetValue] utility, and then converts it to a Java expression
+ * using [JavaValueConverter].
+ */
+internal interface EmptyFieldSpecification {
+
+    /**
+     * Converts Protobuf values to Java expressions.
+     */
+    val converter: JavaValueConverter
```

Comment:
I've used this approach with `UnsetValue` and `JavaValueConverter` just because it was already here for rules.

There is a possible enhancement: since we have an instance of `Field`, we always know its type, and we can go without `equals()` comparing for strings, bytes, repeated and maps. These types can be checked just by their size.

It will drop the need for `JavaValueConverter`, which in turn needs `TypeSystem` to be instantiated.